examples/large_deformation/hyperelastic_tl_up_interactive.py

```diff
@@ -0,0 +1,372 @@
+
+    global_stress.append(stress[0, 0, 0, 0])
+    global_displacement.append(np.max(out['u'].data[:, 0]))
+
+    return out
+
+def main(cli_args):
+    dims = cli_args.dims
+    shape = cli_args.shape
+    centre = cli_args.centre
+    material_parameters = cli_args.material_parameters
+    order = cli_args.order
+
+    ts_vals = cli_args.ts.split(',')
+    ts = {
+        't0' : float(ts_vals[0]), 't1' : float(ts_vals[1]),
+        'n_step' : int(ts_vals[2])}
+
+    do_plot = cli_args.plot
+
+    ### Mesh and regions ###
+    mesh = gen_block_mesh(
+        dims, shape, centre, name='block', verbose=False)
+    domain = FEDomain('domain', mesh)
+
+    omega = domain.create_region('Omega', 'all')
+
+    lbn, rtf = domain.get_mesh_bounding_box()
+    box_regions = define_box_regions(3, lbn, rtf)
+    regions = dict([
+        [r, domain.create_region(r, box_regions[r][0], box_regions[r][1])]
+        for r in box_regions])
+
+    ### Fields ###
+    scalar_field = Field.from_args(
+        'fu', np.float64, 'scalar', omega, approx_order=order-1)
+    vector_field = Field.from_args(
+        'fv', np.float64, 'vector', omega, approx_order=order)
+
+    u = FieldVariable('u', 'unknown', vector_field, history=1)
+    v = FieldVariable('v', 'test', vector_field, primary_var_name='u')
+    p = FieldVariable('p', 'unknown', scalar_field, history=1)
+    q = FieldVariable('q', 'test', scalar_field, primary_var_name='p')
+
+    ### Material ###
+    c10, c01 = material_parameters
+    m = Material(
+        'm', mu=2*c10, kappa=2*c01,
+    )
+
+    ### Boundary conditions ###
+    x_sym = EssentialBC('x_sym', regions['Left'], {'u.0' : 0.0})
+    y_sym = EssentialBC('y_sym', regions['Near'], {'u.1' : 0.0})
+    z_sym = EssentialBC('z_sym', regions['Bottom'], {'u.2' : 0.0})
+    disp_fun = Function('disp_fun', get_displacement)
+    displacement = EssentialBC('displacement', regions['Right'], {'u.0' : disp_fun})
+    ebcs = Conditions([x_sym, y_sym, z_sym, displacement])
+
+    ### Terms and equations ###
+    integral = Integral('i', order=2*order)
+
+    term_neohook = Term.new(
+        'dw_tl_he_neohook(m.mu, v, u)',
+        integral, omega, m=m, v=v, u=u)
+    term_mooney = Term.new(
+        'dw_tl_he_mooney_rivlin(m.kappa, v, u)',
+        integral, omega, m=m, v=v, u=u)
+    term_pressure = Term.new(
+        'dw_tl_bulk_pressure(v, u, p)',
+        integral, omega, v=v, u=u, p=p)
+
+    term_volume_change = Term.new(
+        'dw_tl_volume(q, u)',
+        integral, omega, q=q, u=u, term_mode='volume')
+    term_volume = Term.new(
+        'dw_volume_integrate(q)',
+        integral, omega, q=q)
+
+    eq_balance = Equation('balance', term_neohook+term_mooney+term_pressure)
+    eq_volume = Equation('volume', term_volume_change-term_volume)
+    equations = Equations([eq_balance, eq_volume])
+
+    ### Solvers ###
+    ls = ScipyDirect({})
+    nls_status = IndexedStruct()
+    nls = Newton(
+        {'is_linear' : False},
```

Comment:
The default number of iterations is 1, so you need to increase it here - use, for example,  `'i_max' : 5`, and maybe also tweak the tolerances. On the other hand, `'is_linear' : False` is the default, and is not needed :)